### DISCUSSION TITLE
avoid data loss due to multiple instances writing the same file?

### DISCUSSION THREAD

**Discussion by edrex** [helix-editor/helix#5006] [Ideas] [answered]
**preface:** I'm about a week into switching to helix as my primary editor, motivated by the recent support for marksman, and really enjoying it.

**issue:** I just noticed that some lines were missing from a file, and realized that I had opened the file in multiple instances, and the missing edits had been overwritten.

Some things that would help avoid this type of data loss:
- Warn before writing if file has changed underneath us.
- Option to auto-load changes into a clean buffer on focus events. This would help ensure instances don't diverge.
- Client server architecture (#312) 

File locking is a pain and I'm not suggesting it.

**Comment by kirawi** [comment, marked as answer]
https://github.com/helix-editor/helix/pull/4734

**Reply by edrex** [reply]
Thanks for linking, missed that one in my search
